src/main/java/seedu/address/model/task/TaskMark.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represent a task mark in HR PRO Max++.
+ */
+public class TaskMark {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Task can only be marked as true for completed or false for not completed";
+
+    /**
+     * Accepts only true or false as valid input.
+     */
+    public static final String VALIDATION_REGEX = "(true|false)";
+
+    public final String taskMark;
+
+    /**
+     * Constructs a {@code Mark}.
+     *
+     * @param taskMark A valid description.
+     */
+    public TaskMark(String taskMark) {
+        requireNonNull(taskMark);
+        checkArgument(isValidTaskMark(taskMark), MESSAGE_CONSTRAINTS);
+        this.taskMark = taskMark;
+    }
+
+    /**
+     * Checks if it is a valid TaskMark.
+     * @param test String to check if its a valid TaskMark
+     * @return True if its either true or false.
+     */
+    public static boolean isValidTaskMark(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
+        return taskMark;
+    }
```

Comment:
Maybe we can change this toString to [X] or [ ].

Reply:
Ok changes have been made accordingly